### PR TITLE
Updates related to verified publishers, author/authors

### PR DESCRIPTION
**Pull request by kwalrath**
Staged:
https://sz-www-1.firebaseapp.com/tools/pub/publishing
https://sz-www-1.firebaseapp.com/tools/pub/glossary#uploader
https://sz-www-1.firebaseapp.com/tools/pub/cmd/pub-uploader
https://sz-www-1.firebaseapp.com/tools/pub/pubspec

Contributes to #2029

### CODE REVIEW

File: src/tools/pub/glossary.md [hunk 1]
```diff
@@ -146,16 +146,34 @@ and C are transitive ones.
 
 ## Uploader
```
Comment (mit-mit) [inline src/tools/pub/glossary.md]: Maybe indent this subsection and the verified publisher subsection below under a new overall section?

```
## Managing who has rights to publish

### Uploader

### Verified publisher
```

File: src/tools/pub/glossary.md [hunk 2]
```diff
@@ -146,16 +146,34 @@ and C are transitive ones.
...
+then all members of the publisher can upload the package.
 
 Anyone uploading a new package automatically becomes an uploader for
-that package. Otherwise, to become an uploader, you need to contact an
+that package. Otherwise, to become an uploader of a package that
+doesn't have a verified publisher, you need to contact an
 existing uploader and ask them to add you as another uploader.
 
+
+## Verified publisher
```
Comment (mit-mit) [inline src/tools/pub/glossary.md]: Here, rather than repeating all the details of what a verified publisher is, consider focusing on what members of a publisher have the rights to do (like in the previous Uploaders section). That is:

  - Publish new packages under the publisher
  - Publish new versions of existing packages
  - Edit the publisher metadata

File: src/tools/pub/pubspec.md [hunk 3]
```diff
@@ -172,22 +167,22 @@ The description is plain text: no markdown or HTML.
...
+(`Natalie Weizenbaum <nweiz@google.com>`).
+However, these values weren't verified,
+and pub.dev no longer displays them.
 
-{% prettify yaml %}
-authors:
-- Natalie Weizenbaum <nweiz@google.com>
-- Bob Nystrom <rnystrom@google.com>
-{% endprettify %}
+As of Dart 2.7, the `pub publish` command
```
Comment (mit-mit) [inline src/tools/pub/pubspec.md]: maybe add here: ", and the pub.dev site no longer displays the `author(s)` on a package page.".

File: src/tools/pub/glossary.md (hunk 1 above)
Comment (kwalrath) [reply src/tools/pub/glossary.md]: This is a glossary, not a task oriented doc, so alpha order is the rule.

File: src/tools/pub/glossary.md (hunk 2 above)
Comment (kwalrath) [reply src/tools/pub/glossary.md]: I just stole this definition from pub.dev, since it seemed worthwhile to have the definition on this site too. I didn't want to get into functionality here, since it's easy to miss updating the glossary. Instead I just linked to the verified publishers page. (We don't have any similar page for uploaders, so I guess it makes sense to have the info here.)

(I just checked, and people actually do come to the glossary, yay.)

File: src/tools/pub/pubspec.md (hunk 3 above)
Comment (kwalrath) [reply src/tools/pub/pubspec.md]: I say that up above.

**Comment by kwalrath** [comment]
Thanks for the review, @mit-mit! After my explanations, do you still think I need to make changes?

File: src/tools/pub/glossary.md (hunk 2 above)
Comment (mit-mit) [reply src/tools/pub/glossary.md]: Even if this is the glossary, why the asymmetry between this section (which talks about publishers from a consumers PoV) and the Uploader section (which talks about uploaders from a publisher PoV)?

File: src/tools/pub/pubspec.md (hunk 3 above)
Comment (mit-mit) [reply src/tools/pub/pubspec.md]: Sorry, missed that. I think it would be easier to not miss if we did something like this:

```
You might see an `author` or `authors` section in old pubspecs.
These optional fields were a way to describe
the author(s) of your package and to provide contact information.
Each author could be either a single name
(`Natalie Weizenbaum`) or a name and an email address
(`Natalie Weizenbaum <nweiz@google.com>`).

As of Dart 2.7, pub.dev no longer displays them and the `pub publish` command
displays a warning if your pubspec has an `author` or `authors` section.
```

**Comment by mit-mit** [comment]
See comments

File: src/tools/pub/pubspec.md (hunk 3 above)
Comment (kwalrath) [reply src/tools/pub/pubspec.md]: The thing is, pub.dev no longer displaying the fields is independent of 2.7. But I'll try to reword it to remove that dependency.

File: src/tools/pub/glossary.md (hunk 2 above)
Comment (kwalrath) [reply src/tools/pub/glossary.md]: Funny, I don't see the audiences of the entries as being that different; it's only the degree of completeness that's different, because there's no page dedicated to the uploader concept. The only link to glossary#uploader is from /tools/pub/cmd/pub-uploader, which is intended for publishers; there are no links to glossary#verified-publisher because there's a verified publisher page.

I'm going to cut down the verified publisher entry, since it's only here for completeness.

Comment (kwalrath) [reply src/tools/pub/glossary.md]: I've made the glossary changes (and some others while I'm in there).

If we want to say anything more about publishers, we should say it in one of the other docs, since it's unlikely people will come here to find out about verified publishers.

We might also want to revamp the verified publishers page to be clearer about which sections are for which audiences.

File: src/tools/pub/glossary.md [hunk 4]
```diff
@@ -146,16 +146,34 @@ and C are transitive ones.
...
-same as the [author](/tools/pub/pubspec#authorauthors) of a package.
+Someone who has administrative permissions for a package.
+A package uploader can upload new versions of the package,
+and they can also [add and remove other uploaders](/tools/pub/cmd/pub-uploader)
+for that package. 
+
+If a package has a verified publisher,
+then all members of the publisher can upload the package.
 
 Anyone uploading a new package automatically becomes an uploader for
```
Comment (mit-mit) [inline src/tools/pub/glossary.md]: I would consider removing this paragraph. It repeats the last part of the first paragraph in this section.
